app/src/main/java/ca/dal/cs/athletemonitor/athletemonitor/WorkoutActivity.java

```diff
@@ -0,0 +1,172 @@
+package ca.dal.cs.athletemonitor.athletemonitor;
+
+import android.graphics.Color;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class WorkoutActivity extends AppCompatActivity implements
+        AdapterView.OnItemSelectedListener {
+
+    // Needed private variables
```

Comment:
This should probably not be done this way, the problem is that you are addressing these items throughout the class but you can't be sure at any point that they're non-null. Add a TODO above this block of variables, something like
`// TODO: Refactor these private variables to make code safer, i.e. either make variables local to methods or check for null where necessary`